pkg/apis/core/validation/validation_test.go

```diff
@@ -11964,6 +11986,55 @@ func TestValidateServiceUpdate(t *testing.T) {
 			},
 			numErrs: 1,
 		},
+		/* Service IP Family */
+		{
+			name: "same ServiceIPFamily",
+			tweakSvc: func(oldSvc, newSvc *core.Service) {
+				oldSvc.Spec.Type = core.ServiceTypeClusterIP
+				oldSvc.Spec.ServiceIPFamily = core.IPv4Service
+
+				newSvc.Spec.Type = core.ServiceTypeClusterIP
+				newSvc.Spec.ServiceIPFamily = core.IPv4Service
+			},
+			numErrs: 0,
+		},
+		{
+			name: "ExternalName while changing Service IPFamily",
+			tweakSvc: func(oldSvc, newSvc *core.Service) {
+
+				oldSvc.Spec.ExternalName = "somename"
+				oldSvc.Spec.Type = core.ServiceTypeExternalName
+				oldSvc.Spec.ServiceIPFamily = core.IPv4Service
+
+				newSvc.Spec.ExternalName = "somename"
+				newSvc.Spec.Type = core.ServiceTypeExternalName
+				newSvc.Spec.ServiceIPFamily = core.IPv6Service
+			},
+			numErrs: 0,
+		},
+		{
+			name: "changing to ExternalName while changing ServiceIPFamily",
```

Comment:
why are we testing this?  Why not just make it immutable, period?

Reply:
That test was added before i discover the immutability of the ClusterIP. I will remove it.

Reply:
When i went back to find why the test was there, i found that the original code does not validate immutability in case of `ExternalName` service. This is why it has a test with expected zero error. I will keep the test, unless we see other reasons to remove it.